xarray/backends/zarr.py

```diff
@@ -291,28 +305,17 @@ def encode_zarr_variable(var, needs_copy=True, name=None):
     return var
 
 
-class ZarrStore(AbstractWritableDataStore, DataStorePickleMixin):
+class ZarrStore(AbstractWritableDataStore):
     """Store for reading and writing data via zarr
     """
 
     def __init__(self, store=None, mode='a', synchronizer=None, group=None,
```

Comment:
Following the lead of other DataStore classes like `NetCDF4DataStore`, could we refactor `__init__` to take a zarr group object, and put the version that calls `zarr.open_group` in an `open` classmethod? That retains a bit more flexibility for future use cases.